licence-checker/licence-checker.py

```diff
@@ -52,6 +53,15 @@ def search_line(self, licence_first_word):
     def expected_full_line(self, licence_line):
         return self.comment_prefix + ' ' + licence_line
 
+    def match(self, licence, line, match_regex):
+        """Match a license line wih a retrieved line using either string matching or regex
+        given the comment style.
+        """
+        if match_regex:
+            return re.compile(self.expected_full_line(licence)).match(line)
```

Comment:
Sorry to be nit-picky, but could we move the `re.compile` to `LineCommentStyle.__init__`? It seems better to throw any "this is a bad regex" errors at startup time (and regardless of whether we happen to read a file that needs this comment style).

Reply:
Hmm, that was a terrible suggestion from me! (because neither `licence` nor `match_regex` are known at module load time, when we construct these objects).

I think this would probably be good to change (so we would build the matchers that are currently in COMMENT_CHARS after we've parsed the configuration). But that looks like a bit more engineering: let's do that in a different PR.

The only problem left here is that we need to escape the comment prefix for regexes (because otherwise `/*`, the C-style comment, matches `//` but not `/*`. Oops!). Maybe the best thing is to pass `match_regex` to `search_line` and `expected_full_line` and return a regex from both of them unconditionally? We're also going to have a problem with the implementation of `LicenceHeader.first_word`. How should we do that, do you think?

Reply:
I see the problem. Could we escape the comment style using [`re.escape()`](https://docs.python.org/3/library/re.html#re.escape) and concatenate it with the non-commented license regex line? if okay I'll wait for the new PR to propose an exact solution. Thanks again.

Reply:
Yep, that sounds like the right thing to me.